force-app/main/default/classes/AdvancedSearchController.cls

```diff
@@ -0,0 +1,25 @@
+public with sharing class AdvancedSearchController {
+    @AuraEnabled(cacheable=true)
+    public static list<Title__c> getMovieList(string searchKey) {
+        string searchKeyword = '%' + searchKey + '%';
+		
+       // create movie list to store search result 
+        list<Title__c> movieListRecord = new list<Title__c>();
+     
+	   // query movie records
+        for(Title__c titleObj : [Select id,Name,Director__c,Writer__c,Notable_Actors__c
```

Comment:
Its small, but when writing SOQL you want to follow SQL style best practices to capitalize all of your functions... you did great with the WHERE and the LIKE functions, but you should update this query to have SELECT and FROM.